fix: RegExp "invalid group specifier name" exception on Safari < 16.4

wanderingmeow opened the pull request:
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [X] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [X] **Description:** Provide a concise description of the changes made in this pull request.
- [X] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [X] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [X] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [X] **Testing:** Have you written and run sufficient tests for validating the changes?
- [X] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [X] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

This PR fixed the issue of regex exception for Safari before 16.4 (iOS 16.4), where regex lookbehind syntax is not supported. Before this change, devices that are not supported by or updated to iOS 16.4 or later are unable to use Open WebUI chat, e.g. iPhone 7.

### Fixed

- RegExp "invalid group specifier name" exception on Safari < 16.4 (#3306, #3371)

---

### Additional Information

This issue was first observed in Open WebUI 0.3.5.

For more information, see discussion here: https://github.com/open-webui/open-webui/discussions/3372#discussioncomment-9889979

Prior art:
- https://stackoverflow.com/questions/51568821/works-in-chrome-but-breaks-in-safari-invalid-regular-expression-invalid-group
- https://stackoverflow.com/questions/11761563/javascript-regexp-for-splitting-text-into-sentences-and-keeping-the-delimiter

tjbck commented:
Thanks!

tjbck commented:
This change has been reverted in dev as this broke compatibility with OpenAI models. Feel free to reopen the PR again after THOROUGHLY testing the changes, thanks!

wanderingmeow commented:
Apologize for the oversight in my previous submission. It doesn't handle empty strings properly, which can occur quite often when token generation is slow.

Here's the updated one:
```javascript
let sentences = text.match(/[^\.!\?]+[\.!\?]*|.+/g) || [];
```

I've tested it in both UI and browser console, and it works as expected. Here are the test results:

```javascript
> let text = "Hello, there! What's this? This is a string."
> text.match(/[^\.!\?]+[\.!\?]*|.+/g) || [ "" ]
[ 'Hello, there!', " What's this?", ' This is a string.' ]
> text.split(/(?<=[.!?])\s+/)
[ 'Hello, there!', "What's this?", 'This is a string.' ]
> text = ""
> text.match(/[^\.!\?]+[\.!\?]*|.+/g) || [ "" ]
[ '' ]
> text.split(/(?<=[.!?])\s+/)
[ '' ]
> text = `Multiple
... lines.
... !`
> text.match(/[^\.!\?]+[\.!\?]*|.+/g) || [ "" ]
[ 'Multiple\nlines.', '\n!' ]
> text.split(/(?<=[.!?])\s+/)
[ 'Multiple\nlines.', '!' ]
```

Even though leading whitespace characters exist, they will be trimmed later by `sentence.trim()`.

As this PR has already been merged, I'm unable to reopen it. You could commit this directly. Thank you for your time.
